### ISSUE TITLE
Recursion depth exceeded when using LSTM with dropout and large input dimension

### ISSUE THREAD

**Issue by ameasure** [fchollet/keras#2489]
Fitting a LSTM with an input dimension greater than ~900 results in the following error trace:
https://gist.github.com/ameasure/9f260f9a7a1c1b8552c09aa273879fe0#file-lstm_do_recursion_limit

The error can be recreated with this (it is just the imdb_bidirectional_lstm.py example with 2 minor modifications, LSTM dropout is added and the embedding dimension is increased to 1000): 
https://gist.github.com/ameasure/16fd35b0e409aa7567855e5b04d40788#file-imdb_bidirectional_lstm-py

No problem occurs if the embedding dimension is smaller or no LSTM dropout is used. The problem can also be fixed with:

```
import sys
sys.setrecursionlimit(10000)
```

but there should be a way to do it without having to manually change that limit.

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by braingineer** [comment]
It might be related to this:
https://github.com/Theano/Theano/issues/689

**Comment by nouiz** [comment]
We have someone as part of the GSoC that should work on the recursion problem during the summer. Here is the meta ticket where we try to collect all the problem we know:

https://github.com/Theano/Theano/issues/3607

I added this issue to the list of known issue.

Use the work around until it get fixed.
